react-carousel-component/my-app/src/Carousel.js

```diff
@@ -0,0 +1,81 @@
+import { useState, useEffect } from "react";
+import { FaAngleLeft, FaAngleRight, FaCircle, FaRegCircle, FaRegWindowClose } from "react-icons/fa";
+import './Carousel.css';
+
+export default function Carousel({ images }) {
+  const [currentImageIndex, setCurrentImageIndex] = useState(0);
+
+  const { name: currentName, imageURL: currentImage } = images[currentImageIndex];
+
+  function handleClick(index) {
+    setCurrentImageIndex(index)
+  }
+
+  function handlePrev() {
+    setCurrentImageIndex(((currentImageIndex - 1) + images.length) % images.length)
+  }
+
+  function handleNext() {
+    setCurrentImageIndex((currentImageIndex + 1) % images.length)
+  }
+
+  useEffect(() => {
+    const timeoutID = setTimeout(handleNext, 3000);
+    return () => clearInterval(timeoutID)
+  })
+
+  return (
+    <>
+      <div className="Slideshow">
+        <Button icon="FaAngleLeft" onClick={handlePrev} />
+        <Image url={currentImage} name={currentName} />
+        <Button icon="FaAngleRight" onClick={handleNext} />
+      </div>
+      <Dots numberOfImages={images.length} currentIndex={currentImageIndex} onClick={handleClick} />
+    </>
+  )
+}
+
+function Button({ icon, onClick }) {
+  switch(icon) {
```

Comment:
The problem with this switch is if you decide to use a different Icon then you have to add another switch case. You can actually pass Components as props. Something like this should work:
```
function Button({ icon, onClick }) {
  return <button className="No-style" onClick={onClick}>{icon}</button>;
}
```
and call it with
```
<Button icon={<FaAngleLeft size={70} />} onClick={handlePrev} />
```